main.py

```diff
@@ -2152,18 +2156,14 @@ def do_request(self, prefix, locations):
 
     def format_completion(self, item) -> 'Tuple[str, str]':
         # Sublime handles snippets automatically, so we don't have to care about insertTextFormat.
-        label = item.get("label")
-        detail = item.get("detail")
-        kind = item.get("kind")
+        label = item["label"]
+        detail = item.get("detail") if show_completion_details else None
         if not detail:
-            if kind is not None:
-                detail = completion_item_kind_names[kind]
-        insertText = item.get("insertText", None)
-        if not insertText:
-            insertText = label
+            detail = completion_item_kind_names[item.get("kind", CompletionItemKind.Unknown)]
+        insertText = item.get("insertText") or label
         if insertText[0] == '$':  # sublime needs leading '$' escaped.
             insertText = '\$' + insertText[1:]
-        return "{}\t{}".format(label, detail) if detail else label, insertText
+        return "{} \t {}".format(label, detail), insertText
```

Comment:
Would just one space suffice?

Reply:
If completion label and/or the hint contains spaces and many long items of different length are displayed using 3 spaces seemed to be more readable. The italic style only doesn't cause a good distinction. May be a personal impression. What do you think?

1 space:

![screenshot_before](https://user-images.githubusercontent.com/16542113/31067781-ee4203aa-a754-11e7-86a6-309de5fc5f08.png)


2 spaces:

![screenshot_before](https://user-images.githubusercontent.com/16542113/31067812-1e78d954-a755-11e7-9d33-679dd7dfd76d.png)

Reply:
Agree, 2 is better!